pkg/comp-functions/functions/vshnredis/redis_deploy.go

```diff
@@ -5,27 +5,283 @@ import (
+		},
+	}
+
+	_, _, err = common.CreateMTLSCerts(ctx, comp.GetInstanceNamespace(), comp.GetName(), svc, tlsOpts)
 	if err != nil {
-		err = fmt.Errorf("cannot bootstrap instance namespace: %w", err)
-		return runtime.NewWarningResult(err.Error())
+		return runtime.NewWarningResult(fmt.Errorf("cannot create mTLS certificates: %w", err).Error())
+	}
+
+	if err := createObjectHelmRelease(ctx, comp, svc, secretName); err != nil {
+		return runtime.NewWarningResult(fmt.Errorf("cannot create helm release: %w", err).Error())
+	}
+
+	if err := getConnectionDetails(comp, svc, secretName); err != nil {
+		return runtime.NewWarningResult(fmt.Errorf("cannot set connection details: %w", err).Error())
 	}
 
 	return nil
+}
+
+func createObjectHelmRelease(ctx context.Context, comp *vshnv1.VSHNRedis, svc *runtime.ServiceRuntime, secretName string) error {
+	values, err := newValues(ctx, svc, comp, secretName)
+	if err != nil {
+		return err
+	}
+
+	observedValues, err := common.GetObservedReleaseValues(svc, redisRelease)
+	if err != nil {
+		return fmt.Errorf("cannot get observed release values: %w", err)
+	}
+
+	_, err = maintenance.SetReleaseVersion(ctx, comp.Spec.Parameters.Service.Version, values, observedValues, []string{"image", "tag"})
+	if err != nil {
+		return fmt.Errorf("cannot set redis version for release: %w", err)
+	}
+
+	rel, err := newRelease(ctx, svc, values, comp)
+	if err != nil {
+		return err
+	}
+
+	if err := svc.SetDesiredComposedResourceWithName(rel, redisRelease); err != nil {
+		return err
+	}
+
+	return svc.AddObservedConnectionDetails(redisRelease)
+}
+
+func newValues(ctx context.Context, svc *runtime.ServiceRuntime, comp *vshnv1.VSHNRedis, secret string) (map[string]any, error) {
+	l := svc.Log
+
+	plan := comp.Spec.Parameters.Size.Plan
+	if plan == "" {
+		plan = svc.Config.Data["defaultPlan"]
+	}
+
+	resources, err := utils.FetchPlansFromConfig(ctx, svc, plan)
+	if err != nil {
+		return nil, fmt.Errorf("cannot fetch plans from config: %w", err)
+	}
 
+	vshnSize := comp.GetSize()
+	res, errs := common.GetResources(&vshnSize, resources)
+	if len(errs) != 0 {
+		l.Error(errs[0], "resource calculation error")
+	}
+
+	nodeSelector, err := utils.FetchNodeSelectorFromConfig(ctx, svc, plan, comp.Spec.Parameters.Scheduling.NodeSelector)
+	if err != nil {
+		return nil, err
+	}
+
+	commonConfig := ""
+	if comp.Spec.Parameters.Service.RedisSettings != "" {
+		commonConfig = comp.Spec.Parameters.Service.RedisSettings
+	}
+
+	values := map[string]any{
+		"fullnameOverride": "redis",
+		"architecture":     "standalone",
+
+		"global": map[string]any{
+			"security": map[string]any{
+				"allowInsecureImages": true,
+			},
+		},
+
+		"image": map[string]any{
+			"repository": "bitnami/redis",
+			"tag":        comp.Spec.Parameters.Service.Version,
+		},
+
+		"auth": map[string]any{
+			"enabled":                   true,
+			"existingSecret":            secret,
+			"existingSecretPasswordKey": passwordKey,
+		},
+
+		"tls": map[string]any{
```

Comment:
I am not sure whether TLS is mandatory all the time, isn't it configurable in the claim?

Reply:
You're right, added it back in